### PR TITLE
add vimeo/psalm and run it in CI

### PR DESCRIPTION
**Pull request by alfredbez**
This PR adds [psalm](https://psalm.dev/) with a baseline-file to the CI pipeline and should help to catch new problems.

* In order to get it running, I needed to [replace](https://github.com/OXID-eSales/oxideshop_ce/commit/8fc53ea297da7610b37f0d95c44545093a98a052) the `oxInputException` with the new namespaces class `InputException`.
* Psalm has a built-in fixer, which was able to [fix a few issues in docblocks](https://github.com/OXID-eSales/oxideshop_ce/commit/ca4cbeddce74768d660259930c1a12db1c88c356)
* Psalm found over 10.000 issues, which are all ignored for now via [`psalm-baseline.xml`](https://github.com/OXID-eSales/oxideshop_ce/commit/cf0e3edcb7eba1e742fd63b9eda0539b51f28653). It's almost impossible to fix all of these issues, but that shouldn't be a short-term goal anyhow. Just make sure, that we do not any new issues to the code and fix a few of the existing ones from time to time (boy-scout-rule)

### CODE REVIEW

File: psalm.xml [hunk 1]
```diff
@@ -0,0 +1,151 @@
+<?xml version="1.0"?>
+<psalm
+    totallyTyped="false"
+    resolveFromConfigFile="true"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns="https://getpsalm.org/schema/config"
+    xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
+    errorBaseline="psalm-baseline.xml"
+    autoloader=".psalm-autoload.php"
```
Comment (alfredbez) [inline psalm.xml]: Maybe it is possible to add multiple files here, but I'm not sure how.

**Comment by flow-control** [comment]
Hey @alfredbez,

thank you very much for this pull request and the effort you put into this, as we just merged https://github.com/OXID-eSales/oxideshop_ce/pull/769, could you rebase your PR on latest `master`?
After rebasing, we should get Travis green again ;-)

/Flo

**Comment by flow-control** [comment]
Hey @alfredbez,

I rebased your PR on the latest `master` branch so and solved the conflicts, I will now try my best to get CI :green_heart: again.

/Flo

**Comment by alfredbez** [comment]
Happy to hear that and sorry for not picking this up, I'm currently working on an urgent Corona-related project, that's why I didn't find the time yet.

**Comment by flow-control** [comment]
Hey there :vulcan_salute:

I set the error level to 8 (which is the lowest) and fixed most problems there. We still have a baseline file with some tricky errors, that should be fixed, but the easy fixes are in.
As it was already decided by team to merge this (when CI is :green_heart:), I would suggest we keep the impact low and continue with raising the error level and fixing things from baseline file later :wink: 

Let's see what travis thinks about this

/Flo

**Pull request merged**